helpers/faketask/faketask.go

```diff
@@ -22,46 +22,64 @@ import (
 
 // FakeTask for testing things that require real tasks
 type FakeTask struct {
-	Status string
-	Level  resource.StatusLevel
-	Error  error
+	Status      string
+	Level       resource.StatusLevel
+	Error       error
+	NilAndError bool
```

Comment:
I wonder if it would make more sense to have a `type NilTask struct` that just returns `nil, error` to keep the code clean